src/DataTypes/Serializations/ISerialization.h

```diff
@@ -254,9 +287,48 @@ class ISerialization
     static ColumnPtr getFromSubstreamsCache(SubstreamsCache * cache, const SubstreamPath & path);
 
     static bool isSpecialCompressionAllowed(const SubstreamPath & path);
+
+    template <typename State, typename Serialization>
+    static State * checkAndGetSerializeState(SerializeBinaryBulkStatePtr & state, const Serialization &);
+
+    template <typename State, typename Serialization>
+    static State * checkAndGetDeserializeState(DeserializeBinaryBulkStatePtr & state, const Serialization &);
```

Comment:
I think it shouldn't be in public interface. It is some kind of helper.
Also there are too many templates. `Serialization` is used only to get typeid ...
I would better make this non-static protected method with only `State` template param. Get typeid from `*this` if you want. Or better add `String getName() virtual = 0`.